okio/jvm/src/main/java/okio/Throttler.kt

```diff
@@ -143,6 +162,16 @@ class Throttler internal constructor(
           throw InterruptedIOException("interrupted")
         }
       }
+
+      override suspend fun readAsync(sink: Buffer, byteCount: Long): Long {
+        try {
+          val toRead = takeAsync(byteCount)
+          return super.readAsync(sink, toRead)
+        } catch (e: InterruptedException) {
```

Comment:
Just a thing to be consistent with our blocking APIs, which would otherwise need to throw this checked exception. And yes, we should figure out cancelation.

Reply:
Ah right, it's a checked exception. I think coroutine cancellation should just work.